### DISCUSSION TITLE
How to use Networking autogenerated from open api swagger instead of Nylos?

### DISCUSSION THREAD

**Discussion by abbjetmus** [nylo-core/nylo#83] [Q&A] [answered]
Hi!
I'm using https://github.com/epam-cross-platform-lab/swagger-dart-code-generator to generate networking, but I'm not sure how make it fit in Nylo instead of Nylos networking. Was thinking of adding a prodvider, but I want a global instance of 

var api = Swagger.create(baseUrl: baseUrl); 

and access it everywhere.

```dart

@ChopperApi()
abstract class Swagger extends ChopperService {
  static Swagger create({
    ChopperClient? client,
    http.Client? httpClient,
    Authenticator? authenticator,
    Converter? converter,
    Uri? baseUrl,
    Iterable<dynamic>? interceptors,
  }) {
    if (client != null) {
      return _$Swagger(client);
    }

    final newClient = ChopperClient(
        services: [_$Swagger()],
        converter: converter ?? $JsonSerializableConverter(),
        interceptors: interceptors ?? [],
        client: httpClient,
        authenticator: authenticator,
        baseUrl: baseUrl ?? Uri.parse('http://'));
    return _$Swagger(newClient);
  }

  ///
  ///@param id
  Future<chopper.Response<List<CompanyBooking>>> apiBookingAllCompanyPlaceIdGet(
      {required int? id}) {
    generatedMapping.putIfAbsent(
        CompanyBooking, () => CompanyBooking.fromJsonFactory);

    return _apiBookingAllCompanyPlaceIdGet(id: id);
  }

```

What is the best way of fitting this in Nylos context?

**Comment by agordn52** [comment, marked as answer]
Hi @abbjetmus,

I think the best way would be to convert your Swagger schema into Postman and then use Nylo's CLI to import the Postman file.
https://learning.postman.com/docs/getting-started/importing-and-exporting/importing-from-swagger/

You can use 
`metro make:api_service booking_api_service -p "your-postman-file"` to auto-create all the API services.

I'll attach a reference to the docs at my earliest convenience.
